File: app_xsg/src/main/java/com/sc/app_xsg/receiver/StartReceiver.java
package com.sc.app_xsg.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.sc.app_xsg.activity.MainActivity;
import com.sc.app_xsg.app.AppHope;
import timber.log.Timber;

/**
 * @author tsc
 * @version 0.0.0-1
 * @date 2022/8/30 15:30
 * @description
 */
public class StartReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() == Intent.ACTION_BOOT_COMPLETED) {
            Timber.i(AppHope.TAG + " ACTION_BOOT_COMPLETED");
            Intent intent1 = new Intent(context, MainActivity.class);
            intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(intent1);
        }
    }
}

File: app_xw/src/main/java/com/sc/xwservice/receiver/StartReceiver.kt
package com.sc.xwservice.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.sc.xwservice.app.App.Companion.TAG
import timber.log.Timber

/**
 * @author  tsc
 * @date  2022/7/30 15:08
 * @version 0.0.0-1
 * @description
 */
class StartReceiver : BroadcastReceiver() {
    override fun onReceive(p0: Context?, p1: Intent?) {
        if (p1?.action == Intent.ACTION_BOOT_COMPLETED) {
            Timber.i("$TAG 开机自启动了")
        }
    }
}